test.js

```diff
@@ -55,11 +55,25 @@ describe('ua-bruteforce', function () {
     tablet: { result: true, tablet: true },
     desktop: { result: false }
   }
+  // The same user-agent string belongs to both `desktop` and `mobile` type entries. No chance to detect `deviceType` properly.
+  // https://github.com/intoli/user-agents/blob/867e318bc00880ae00437e5e8efaa8e5e7ac0696/src/user-agents.json.gz
+  // user-agents v1.0.843
+  const exclude =
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
+
+  const ua = deviceCategory =>
+    ua[deviceCategory] ||
+    (ua[deviceCategory] = new UserAgent([
+      ({ userAgent }) => userAgent !== exclude,
+      { deviceCategory }
+    ]))
```

Comment:
I don't think we need to cache the `UserAgent` instances here

Reply:
Also, I'm having trouble looking up the `UserAgent` API, do you know where its repo is located now?

Reply:
This just follows the guide: https://github.com/intoli/user-agents#useragentrandom. `random()` generates new instances too, but also reuses filtered data preset and so that's «over 100x faster than the initial construction».

Reply:
ah sorry I thought we were using `user-agent` not `user-agents`

Reply:
Are the tests too slow without this optimization?

Reply:
I can simplify a bit to make it clearer

Reply:
premature optimization is the root of a lot of evil :P